File: frontEnd/src/types/dto/stepDTO/WhileStepDTO.ts
// while循环操作
import {StepBaseDTO} from "@/types/dto/base/StepBaseDTO";
import {ConditionalRelationship} from "@/types/enum/condation/ConditionalRelationship";
import {AssertStepDTO} from "@/types/dto/stepDTO/AssertStepDTO";

export class WhileStepDTO extends StepBaseDTO {
    stepType: string = 'WHILE';

    // 条件关系
    conditionalRelationship: ConditionalRelationship | string = "AND";

    // 条件列表
    conditionList: AssertStepDTO[] = [];

    // 最大循环次数
    maxLoopCount: number = 0;
}

File: backEnd/platform-automation/src/main/java/com/mokatest/platform/demos/domain/ui/dto/step/WhileStepDTO.java
package com.mokatest.platform.demos.domain.ui.dto.step;// while循环操作


import com.mokatest.platform.demos.domain.ui.uiEnum.condation.ConditionalRelationship;
import lombok.Data;

import java.util.List;

@Data
public class WhileStepDTO extends StepBaseDTO {
    private String stepType;

    // 条件关系
    private ConditionalRelationship conditionalRelationship;

    // 条件列表
    private List<AssertStepDTO> conditionList;

    // 最大循环次数
    private Integer maxLoopCount;
}






